### PR TITLE
Add error handling basics

### PR DESCRIPTION
**Pull request by Mexolius**

### CODE REVIEW

**Comment by grzanka** [comment]
@Mexolius please fix conflicts

**Comment by Mexolius** [comment]
To view implemented solution place non-numeric or incorrect (too big/small) input in the calculator tab. Also you can now input fractoins in the input and it won't result in a crash due to libdedx trying to calculate for value 0

**Comment by grzanka** [comment]
Fix conflicts
image

**Comment by grzanka** [comment]
shot-2022-01-03_21-52-49

**Comment by Mexolius** [comment]
And that's the new functionality, at least we see it works rather well. More seriously - must be some post-merge bug - will fix soon

**Comment by grzanka** [comment]
The error message seems a bit unrelated:

image

**Comment by Mexolius** [comment]
The error message is what libdEdx provided

**Comment by Mexolius** [comment]
Actually, we could make a gracefull recovery from this case - all negative numbers can be treated the same way NaNs are right now - with a different message

**Comment by grzanka** [comment]
> Actually, we could make a gracefull recovery from this case - all negative numbers can be treated the same way NaNs are right now - with a different message

As first approach I'd simply test if `energy > 0`. More advanced error checking could you min and max values of energies which could be extracted from dedx, but this is far from the scope of current PR.

File: src/Components/Pages/Data/Calculator.js [hunk 1]
```diff
@@ -38,7 +40,9 @@ function toNumericInput(inputArray, separator){
 function transformInputs(inputArray, separator){
     const numericInputs = toNumericInput(inputArray, separator)
     if(numericInputs.some(ni => isNaN(ni))){
-        throw EvalError("Some of the inputs are non-numeric.")
+        throw new NonNumericError()
+    } else if (numericInputs.some(ni => ni < 0)){
```
Comment (grzanka) [inline src/Components/Pages/Data/Calculator.js:44]: Can it be `ni <= 0` ?

Comment (Mexolius) [reply src/Components/Pages/Data/Calculator.js:44]: 0 is not an error - in fact it always appears in the input when trying to type numbers smaller than 1. In other words you always have to have a standalone 0 before you have a 0.5. If it was an error you wouldn't be able to input those values

Comment (Mexolius) [reply src/Components/Pages/Data/Calculator.js:44]: One thing we didn't think of earlier would be updating the results only when the new line would be added - this way standalone 0 could be treated as an error. That change includes somewhat major refactoring so i wouldn't want to do it in this PR tho.

**Review by grzanka — APPROVED** [review]
LGTM

**Pull request merged**